test/python/compiler/test_assembler.py

```diff
@@ -1184,6 +1184,72 @@ def test_assemble_single_instruction(self):
         qobj = assemble(inst, self.backend)
         validate_qobj_against_schema(qobj)
 
+    def test_assemble_disjoint_time(self):
+        """Test that assembly errors when qubits are measured in disjoint time."""
+
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        with self.assertRaises(QiskitError):
+            assemble(schedule,
+                     qubit_lo_freq=self.default_qubit_lo_freq,
+                     meas_lo_freq=self.default_meas_lo_freq,
+                     meas_map=[[0, 1]])
+
+    def test_assemble_non_disjoint_time_single_meas_map(self):
+        """Test that assembly works when qubits are measured in non-disjoint time
+        within the same measurement map list."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 5,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0, 1]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_non_disjoint_time(self):
+        """Test that assembly works when qubits are measured in disjoint time."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0], [1]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_invalid_qubits(self):
+        """Test that assembly errors when qubits that are not in the measurement map
+        is measured."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(2), MemorySlot(2)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(3), MemorySlot(3)),
+        )
+        with self.assertRaises(QiskitError):
+            assemble(schedule,
+                     qubit_lo_freq=self.default_qubit_lo_freq,
+                     meas_lo_freq=self.default_meas_lo_freq,
+                     meas_map=[[0, 1, 2]])
```

Comment:
Check that this is also fine?
```suggestion
                     meas_map=[[0, 1, 2], [3]])
```

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/65a3fc296ab5ed9747ef3df78aaf1f82e7741f7c